source/loader/ur_ldrddi.cpp

```diff
@@ -2601,14 +2540,6 @@ __urdlllocal ur_result_t UR_APICALL urProgramGetNativeHandle(
         return result;
     }
 
-    try {
-        // convert platform handle to loader handle
-        *phNativeProgram = reinterpret_cast<ur_native_handle_t>(
```

Comment:
@bmyates : why is that we need this kind of logic in the L0 loader, of translating driver handles to L0 handles? Do you see any risks on removing this code in the UR loader?

Reply:
The loader handles are there to find the appropriate adapter-level function to call. The alternative is to either make all functions accept an adapter handle as one of the arguments or create a way to map normal handles to adapter handles.

This mechanism is not used when there's only one adapter, see https://github.com/oneapi-src/unified-runtime/issues/355.

Reply:
> The alternative is to either make all functions accept an adapter handle as one of the arguments or create a way to map normal handles to adapter handles.

There's no need to. Most interop calls accept a context, which has the right adapter, and this patch exactly changes the behavior to find the adapter in the context, rather the native handle. The only notable exception is platform: https://github.com/oneapi-src/unified-runtime/issues/1068. That one certainly needs an adapter as an argument.

Reply:
> There's no need to.

I know, I agree. I was replying to Jamie's question.

Reply:
thanks @pbalcer , @alexbatashev . Ah, this is only for native handles. That's ok then.

+1